seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java

```diff
@@ -107,6 +108,10 @@ public void write(SeaTunnelRow seaTunnelRow) throws FileConnectorException {
         if (currentBatchSize >= batchSize) {
             this.partId++;
             currentBatchSize = 0;
+            if (beingWrittenFile.size() == 1) {
```

Comment:
Hi, why use `==` not `>=`?

Reply:
Seem like you want find filePath belong to this seatunnelRow, maybe you can move this logic to `OrcWriteStrategy`, because only `OrcWriteStrategy` use `writtenFileStatusMap`.
Also you can change `writtenFileStatusMap` to hashset, because value alway be true.

Reply:
> Hi, why use `==` not `>=`?

actually, the size of beingWrittenFile should be always 1 at any time while processing data. (if not, the code I changed should not worked)

Reply:
> Seem like you want find filePath belong to this seatunnelRow, maybe you can move this logic to `OrcWriteStrategy`, because only `OrcWriteStrategy` use `writtenFileStatusMap`. Also you can change `writtenFileStatusMap` to hashset, because value alway be true.

yes, I will make some code update and recommit it in this weekend.

Reply:
> > Hi, why use `==` not `>=`?
> 
> actually, the size of beingWrittenFile should be always 1 at any time while processing data. (if not, the code I changed should not worked)

I'm not sure about that if you configure partition key.

Reply:
Code  is updated now.

- HashMap was changed to HashSet.
- the `size == 1` logic is to avoid any confluence may made by my code.

`writtenFileStatusSet` need add the finished file name before the `beingWrittenFile` clear in `AbstractWriteStrategy` class, the file close logic actually was wrote into  `OrcWriteStrategy`.

Reply:
@TyrantLucifer PTAL